std/json.d

```diff
@@ -96,11 +96,86 @@ enum JSON_TYPE : byte
     FALSE    /// ditto
 }
 
+private struct DefaultPolicy
+{
+    enum preserveMemberOrder = false;
+}
+private struct PreserveMemberOrderPolicy
+{
+    enum preserveMemberOrder = true;
+}
+
+struct OrderedAA(K,V)
+{
+    struct Member
+    {
+        K name;
+        V value;
+    }
+    private Member[] members;
+    size_t length() const { return members.length; }
+    inout(V)* opIn(in char[] name) inout
```

Comment:
char[] instead of K ? (do we need specialization for case K==string to allow char[] in that case?)
=> a unitest with orderedAA!(int, double) would've caught this bug